Source/ProgressionSystemRuntime/Private/Components/PSSpotComponent.cpp

```diff
@@ -47,6 +47,50 @@ void UPSSpotComponent::OnReset_Implementation()
 	}
 }
 
+// Listen game states to switch character skin. 
+void UPSSpotComponent::OnGameStateChanged_Implementation(ECurrentGameState CurrentGameState)
+{
+	// it's possible that spot might not be loaded till that time so no ensure added
+	UPSSpotComponent* CurrentSpot = UPSWorldSubsystem::Get().GetCurrentSpot();
+
+	// check the spot  
+	if (CurrentSpot == nullptr || CurrentSpot != this)
+	{
+		return;
+	}
+
+	if (CurrentGameState == ECurrentGameState::GameStarting)
+	{
+		UMySkeletalMeshComponent& MeshComp = GetMeshChecked();
+		const int32 CurrentSkinIndex = MeshComp.GetAppliedSkinIndex();
+		bool isCurrentSkinAvailable = MeshComp.IsSkinAvailable(CurrentSkinIndex);
+
+		if (!isCurrentSkinAvailable)
```

Comment:
please use early returns for current and previous conditions to decrease nesting

Reply:
done https://github.com/moinm3uw/ProgressionSystem/pull/63/commits/f9f229af0b50920274b6665491f2a51b2f9c910a